reconcile/utils/openshift_resource.py

```diff
@@ -633,41 +646,54 @@ def add_desired(
             ]
             admin_token_usage[name] = privileged
 
-    def get_desired(self, cluster, namespace, resource_type, name):
+    def get_desired(
+        self, cluster: str, namespace: str, resource_type: str, name: str
+    ) -> OpenshiftResource | None:
         try:
             return self._clusters[cluster][namespace][resource_type]["desired"][name]
         except KeyError:
             return None
 
-    def get_desired_by_type(self, cluster, namespace, resource_type):
+    def get_desired_by_type(
+        self, cluster: str, namespace: str, resource_type: str
+    ) -> dict[str, OpenshiftResource] | None:
         try:
             return self._clusters[cluster][namespace][resource_type]["desired"]
         except KeyError:
             return None
 
-    def get_current(self, cluster, namespace, resource_type, name):
+    def get_current(
+        self, cluster: str, namespace: str, resource_type: str, name: str
+    ) -> OpenshiftResource | None:
         try:
             return self._clusters[cluster][namespace][resource_type]["current"][name]
         except KeyError:
             return None
 
-    def add_current(self, cluster, namespace, resource_type, name, value):
+    def add_current(
+        self,
+        cluster: str,
+        namespace: str,
+        resource_type: str,
+        name: str,
+        value: OpenshiftResource,
+    ) -> None:
         with self._lock:
             current = self._clusters[cluster][namespace][resource_type]["current"]
             current[name] = value
 
-    def __iter__(self):
+    def __iter__(self) -> Generator[tuple[str, str, str, dict[str, Any]], None, None]:
```

Comment:
here can use `Iterator`

Reply:
Good spot. Fixed in [de852f7](https://github.com/app-sre/qontract-reconcile/pull/5169/commits/de852f7e3a716235e1c61a3889ff1905e39f7014)